sympy/polys/domains/rationalfield.py

```diff
@@ -159,5 +159,21 @@ def denom(self, a):
         """Returns denominator of ``a``. """
         return a.denominator
 
+    def is_square(self, a):
+        """Return ``True`` if ``a`` is a perfect square. """
+        return is_square(a.numerator) and is_square(a.denominator)
+
+    def exsqrt(self, a):
+        """Compute the non-negative exact square root of ``a`` if it is a
+        perfect square, or return ``None`` if it isn't. """
```

Comment:
Single line summary in a docstring should just be a single line and we should avoid using contractions like "isn't".

I would say
```
Non-negative square root of `a` if `a` is square or otherwise None.

See also
========

is_square
```
Then `is_square` docstring can clarify what it means for a rational to be square.